smart-contracts/contracts/airdrop/src/admin.rs

```diff
@@ -0,0 +1,243 @@
+                if contract_bank_balance < config.airdrop_amount {
+                    return Err(AirdropErrors::InvalidChangeInConfig {});
+                }
+            }
+        }
+    }
+
+    // Save the new airdrop configuration to storage
+    AIRDROP_CONFIG.save(deps.storage, &config)?;
+
+    // Return a default response to indicate success
+    // TODO: Add events
+    Ok(Response::default())
+}
+
+pub fn execute_add_users(
+    deps: DepsMut,
+    users: Vec<String>,
+    amounts: Vec<Uint128>,
+) -> Result<Response, AirdropErrors> {
+    // Load the current airdrop configuration from storage
+    let current_airdrop_config = AIRDROP_CONFIG.load(deps.storage)?;
+
+    // Check if the current airdrop window is not open (start_height or end_height not zero)
+    if current_airdrop_config.start_height != 0 || current_airdrop_config.end_height != 0 {
+        return Err(AirdropErrors::InvalidChangeUserInfo {});
+    }
+
+    // Check if the number of users and amounts provided match
+    if users.len() != amounts.len() {
+        return Err(AirdropErrors::Std(StdError::GenericErr {
+            msg: "Deposit amount weight for primitive is zero".to_string(),
+        }));
+    }
+
+    // Loop through the provided users and amounts
+    for number in 0..=users.len() {
+        // Validate the user's address
+        deps.api.addr_validate(&users[number].to_string())?;
+
+        // Validate that the amount is not negative
+        if amounts[number] < Uint128::zero() {
+            return Err(AirdropErrors::Std(StdError::GenericErr {
+                msg: "Amount at index :".to_string()
+                    + &*number.to_string()
+                    + &*"is negative".to_string(),
+            }));
+        }
+
+        // Load the user's current information from storage
+        let user_info = USER_INFO.load(deps.storage, users[number].clone())?;
+
+        // Check if the user has not claimed and has no existing claimable amount
+        if user_info.get_claimable_amount() == Uint128::zero() && !user_info.get_claimed_flag() {
+            // Save the user's information with the given info
+            USER_INFO.save(deps.storage, users[number].clone(), &user_info)?;
+        }
+    }
+
+    // Calculate the total claimable amount from USER_INFO
+    let total_in_user_info = get_total_in_user_info(deps.storage);
+
+    // Check if the total claimable amount exceeds the airdrop amount
+    if total_in_user_info > current_airdrop_config.airdrop_amount {
+        return Err(AirdropErrors::Std(StdError::GenericErr {
+            msg: "Total amount in the given user amounts".to_string()
+                + &*total_in_user_info.to_string()
+                + &*" is greater than ".to_string()
+                + &*current_airdrop_config.airdrop_amount.to_string(),
+        }));
+    }
+
+    // Return a default response if all checks pass
+    // TODO: Add events
+    Ok(Response::default())
+}
+
+pub fn execute_set_users(
```

Comment:
Does this require us to set a new airdrop config set in the future and then seperately update the user set?

Reply:
New airdrop would mean a new contract.